File: packages/shared/src/enums.ts
export enum UserRole {
  Farmer = "farmer",
  Consumer = "consumer",
  Rider = "rider",
}

export enum OrderStatus {
  Pending = "pending",
  Matched = "matched",
  PickedUp = "picked_up",
  InTransit = "in_transit",
  Delivered = "delivered",
  Cancelled = "cancelled",
  Disputed = "disputed",
}

export enum TripStatus {
  Scheduled = "scheduled",
  InTransit = "in_transit",
  Completed = "completed",
  Cancelled = "cancelled",
}

export enum PaymentMethod {
  Cash = "cash",
  ESewa = "esewa",
  Khalti = "khalti",
  ConnectIPS = "connectips",
}

export enum PaymentStatus {
  Pending = "pending",
  Escrowed = "escrowed",
  Collected = "collected",
  Settled = "settled",
  Refunded = "refunded",
}

export enum VehicleType {
  Bike = "bike",
  Car = "car",
  Truck = "truck",
  Bus = "bus",
  Other = "other",
}

export enum RoleRated {
  Farmer = "farmer",
  Consumer = "consumer",
  Rider = "rider",
}

export enum NotificationCategory {
  OrderMatched = "order_matched",
  RiderPickedUp = "rider_picked_up",
  RiderArriving = "rider_arriving",
  OrderDelivered = "order_delivered",
  NewOrderForFarmer = "new_order_for_farmer",
  RiderArrivingForPickup = "rider_arriving_for_pickup",
  NewOrderMatch = "new_order_match",
  TripReminder = "trip_reminder",
  DeliveryConfirmed = "delivery_confirmed",
}

export enum DevicePlatform {
  Web = "web",
  Android = "android",
  Ios = "ios",
}

export enum OrderItemStatus {
  PendingPickup = "pending_pickup",
  PickedUp = "picked_up",
  Unavailable = "unavailable",
}

export enum PayoutStatus {
  Pending = "pending",
  Settled = "settled",
  Refunded = "refunded",
}

export enum PingStatus {
  Pending = "pending",
  Accepted = "accepted",
  Declined = "declined",
  Expired = "expired",
}

File: apps/mobile/lib/core/enums.dart
// Application enums matching the database enum types.

enum UserRole {
  farmer,
  consumer,
  rider;

  String get label {
    switch (this) {
      case UserRole.farmer:
        return 'Farmer';
      case UserRole.consumer:
        return 'Consumer';
      case UserRole.rider:
        return 'Rider';
    }
  }

  static UserRole fromString(String value) {
    return UserRole.values.firstWhere(
      (e) => e.name == value,
      orElse: () => UserRole.consumer,
    );
  }
}

enum OrderStatus {
  pending,
  matched,
  pickedUp,
  inTransit,
  delivered,
  cancelled,
  disputed;

  String get dbValue {
    switch (this) {
      case OrderStatus.pickedUp:
        return 'picked_up';
      case OrderStatus.inTransit:
        return 'in_transit';
      default:
        return name;
    }
  }

  String get label {
    switch (this) {
      case OrderStatus.pending:
        return 'Pending';
      case OrderStatus.matched:
        return 'Matched';
      case OrderStatus.pickedUp:
        return 'Picked Up';
      case OrderStatus.inTransit:
        return 'In Transit';
      case OrderStatus.delivered:
        return 'Delivered';
      case OrderStatus.cancelled:
        return 'Cancelled';
      case OrderStatus.disputed:
        return 'Disputed';
    }
  }

  static OrderStatus fromDb(String value) {
    switch (value) {
      case 'picked_up':
        return OrderStatus.pickedUp;
      case 'in_transit':
        return OrderStatus.inTransit;
      default:
        return OrderStatus.values.firstWhere(
          (e) => e.name == value,
          orElse: () => OrderStatus.pending,
        );
    }
  }

  bool get isActive =>
      this == pending || this == matched || this == pickedUp || this == inTransit;
}

enum TripStatus {
  scheduled,
  inTransit,
  completed,
  cancelled;

  String get dbValue {
    switch (this) {
      case TripStatus.inTransit:
        return 'in_transit';
      default:
        return name;
    }
  }

  String get label {
    switch (this) {
      case TripStatus.scheduled:
        return 'Scheduled';
      case TripStatus.inTransit:
        return 'In Transit';
      case TripStatus.completed:
        return 'Completed';
      case TripStatus.cancelled:
        return 'Cancelled';
    }
  }

  static TripStatus fromDb(String value) {
    switch (value) {
      case 'in_transit':
        return TripStatus.inTransit;
      default:
        return TripStatus.values.firstWhere(
          (e) => e.name == value,
          orElse: () => TripStatus.scheduled,
        );
    }
  }

  bool get isActive => this == scheduled || this == inTransit;
}

enum PaymentMethod {
  cash,
  esewa,
  khalti;

  String get label {
    switch (this) {
      case PaymentMethod.cash:
        return 'Cash on Delivery';
      case PaymentMethod.esewa:
        return 'eSewa';
      case PaymentMethod.khalti:
        return 'Khalti';
    }
  }

  static PaymentMethod fromString(String value) {
    return PaymentMethod.values.firstWhere(
      (e) => e.name == value,
      orElse: () => PaymentMethod.cash,
    );
  }
}

enum PaymentStatus {
  pending,
  escrowed,
  collected,
  settled,
  refunded;

  static PaymentStatus fromString(String value) {
    return PaymentStatus.values.firstWhere(
      (e) => e.name == value,
      orElse: () => PaymentStatus.pending,
    );
  }
}

enum VehicleType {
  bike,
  car,
  truck,
  bus,
  other;

  String get label {
    return name[0].toUpperCase() + name.substring(1);
  }

  static VehicleType fromString(String value) {
    return VehicleType.values.firstWhere(
      (e) => e.name == value,
      orElse: () => VehicleType.other,
    );
  }
}
